ordered_product/tests/test_ordered_product_page.py

```diff
@@ -0,0 +1,278 @@
+        response = client.get('/ordered_product/orders/'+str(ordered_product0.id)+',delete')
+        assert ordered_product0 not in OrderedProduct.objects.filter(user_name=saved_client0)
+        assert response.status_code == 302
+        assert response['Location'] == '/client/'
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert template_names == set()
+
+    @pytest.mark.django_db()
+    def test_display_new_order_page(self, client, client0_login):
+        response = client.get('/ordered_product/new_order/')
+        assert response.status_code == 200
+        template_names = set(template.origin.template_name for template in response.templates)
+        assert 'ordered_product/new_order.html' in template_names
+
+    @pytest.mark.django_db()
+    def test_search_product_by_product_name(self, client, saved_supplier_product0, client0_login):
+        VALID_PRODUCT_NAME = saved_supplier_product0.qr_code.product_name
+        product = Product.objects.get(product_name=VALID_PRODUCT_NAME)
+        supplier_products_by_objects_filter = SupplierProduct.objects.filter(qr_code=product)
+        assert saved_supplier_product0 in supplier_products_by_objects_filter
+        response = client.get(
+            '/ordered_product/new_order/search_product_by_product_name',
+            {'product_name': VALID_PRODUCT_NAME}
+        )
+        assert response.status_code == 200
+        assert response.context['message'] == ""
+        supplier_products_by_the_web_view = response.context['supplier_products']
+        assert list(supplier_products_by_the_web_view) == list(supplier_products_by_objects_filter)
+
+    @pytest.mark.django_db()
+    def test_search_product_by_location(
+        self,
+        client,
+        saved_supplier_product0,
+        saved_client0,
+        delivery_location2,
+        client0_login
+    ):
+        delivery_location2.save()
+        assert delivery_location2.location == saved_client0.area
+        assert delivery_location2.user_name == saved_supplier_product0.user_name
+
+        deliveries_in_client_location = DeliveryLocation.objects.filter(location=saved_client0.area)
+        assert delivery_location2 in deliveries_in_client_location
+        supplier_products_by_objects_filter = []
+        for delivery in deliveries_in_client_location:
+            supplier_products_by_objects_filter += SupplierProduct.objects.filter(user_name=delivery.user_name)
+        assert saved_supplier_product0 in supplier_products_by_objects_filter
+
+        response = client.get('/ordered_product/new_order/search_product_by_location')
+        assert response.status_code == 200
+        assert response.context['message'] == ""
+        supplier_products_by_the_web_view = response.context['supplier_products']
+        assert list(supplier_products_by_the_web_view) == list(supplier_products_by_objects_filter)
```

Comment:
I recommend using python's `sorted` to wrap both sides of the equation, it'll remove flakiness from the test.

Reply:
Done.